clients/imodelhub/src/imodelbank/IModelBankBasicAuthorizationClient.ts

```diff
@@ -2,69 +2,28 @@
 * Copyright (c) Bentley Systems, Incorporated. All rights reserved.
 * See LICENSE.md in the project root for license terms and full copyright notice.
 *--------------------------------------------------------------------------------------------*/
-import { AuthStatus, BeEvent, BentleyError, ClientRequestContext, Logger } from "@bentley/bentleyjs-core";
+import { BeEvent, ClientRequestContext } from "@bentley/bentleyjs-core";
 import { FrontendAuthorizationClient } from "@bentley/frontend-authorization-client";
-import { AccessToken, IncludePrefix, ITwinClientLoggerCategory, TokenPrefix, UserInfo } from "@bentley/itwin-client";
-const loggerCategory = ITwinClientLoggerCategory.Authorization;
+import { AccessTokenString } from "@bentley/itwin-client";
 
-/**
- * Implements AccessToken that uses Basic access authentication
- * @internal
- */
-@TokenPrefix("Basic")
-export class BasicAccessToken extends AccessToken {
-
-  constructor(tokenStr?: string) {
-    super(tokenStr, undefined, undefined, undefined);
-    this.setPrefix("Basic");
-  }
-  /**
-   * Create a BasicAccessToken from user credentials
-   * @param userCredentials User credentials containing email and password of the user.
-   */
-  public static fromCredentials(userCredentials: any): AccessToken {
-    const basicToken = new BasicAccessToken("");
-    basicToken._tokenString = Buffer.from(`${userCredentials.email}:${userCredentials.password}`).toString("base64");
-    return basicToken;
-  }
-  /**
-   * Creates a token to be used in Authorization header.
-   * @param includePrefix Set to Yes if prefix (Basic) should be included before the token.
-   */
-  public override toTokenString(includePrefix: IncludePrefix = IncludePrefix.Yes): string {
-    let token: string = "";
-    if (includePrefix === IncludePrefix.Yes)
-      token += `${this._prefix} `;
-
-    token += this._tokenString;
-    return token;
-  }
-  /**
-   * initialize the tokenString field of the current instance of BasicAccessToken
-   * @param tokenStr String representation of the token
-   */
-  public override initFromTokenString(tokenStr: string): void {
-    if (!tokenStr.startsWith(this._prefix)) {
-      throw new BentleyError(AuthStatus.Error, "Invalid access token", Logger.logError, loggerCategory, () => ({ tokenStr }));
-    }
-    const userPass = tokenStr.substr(this._prefix.length + 1);
-    this._tokenString = userPass;
-  }
+export function tokenFromUserCredentials(userCredentials: any): AccessTokenString {
+  const tokenString = Buffer.from(`${userCredentials.email}:${userCredentials.password}`).toString("base64");
+  return tokenString;
 }
 
 /** Implements the user permission abstraction by creating a BasicAccessToken. Note that the corresponding IModelBank server must
  * be able to tolerate this BasicAccessToken.
  * @internal
  */
 export class IModelBankBasicAuthorizationClient implements FrontendAuthorizationClient {
```

Comment:
@Animagne do we intend to continue supporting basic authentication for the Bank or is the requirement going forward that you will be required to use OAuth2?

Reply:
I'm not aware of any changes in that requirement. Basic authentication is just a stand in for any other kind of authentication that's not based on OIDC.